mongodbatlas/fw_resource_mongodbatlas_database_user_test.go

```diff
@@ -0,0 +1,281 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	fwresource "github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func TestMongoDBAtlasDatabaseUserResource_Schema(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	schemaRequest := fwresource.SchemaRequest{}
+	schemaResponse := &fwresource.SchemaResponse{}
+
+	// Instantiate the resource.Resource and call its Schema method
+	NewDatabaseUserRS().Schema(ctx, schemaRequest, schemaResponse)
+
+	if schemaResponse.Diagnostics.HasError() {
+		t.Fatalf("Schema method diagnostics: %+v", schemaResponse.Diagnostics)
+	}
+
+	// Validate the schema
+	diagnostics := schemaResponse.Schema.ValidateImplementation(ctx)
+
+	if diagnostics.HasError() {
+		t.Fatalf("Schema validation diagnostics: %+v", diagnostics)
+	}
+}
```

Comment:
This unit test validates the resource schema as explained in the [Hashicorp doc](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/schemas#unit-testing)